### PR TITLE
fix Windows test issues

### PR DESCRIPTION
**Pull request by fritzmg**

### CODE REVIEW

File: core-bundle/src/Doctrine/Backup/Backup.php [hunk 1]
```diff
@@ -94,6 +95,6 @@ private static function validateFilePath(string $filepath): string
             throw new BackupManagerException(sprintf('The filepath "%s" does not match "%s"', $filepath, self::VALID_BACKUP_NAME_REGEX));
         }
 
-        return $filepath;
+        return Path::normalize($filepath);
```
Comment (fritzmg) [inline core-bundle/src/Doctrine/Backup/Backup.php:98]: The backup manager receives a base path with forward slashes, but Symfony Finder will then return the file path with backward slashes under Windows (for the path part below the base path). We can either fix this by normalising the file paths before comparing them in the test - or we normalize them here. I think I would go for the latter.

File: core-bundle/tests/Doctrine/Backup/BackupTest.php [hunk 2]
```diff
@@ -83,6 +84,6 @@ public function invalidFileNameProvider(): \Generator
 
     private function getValidBackupPath(): string
     {
-        return $this->getTempDir().'/valid_backup_filename__20211101141254.sql';
+        return Path::join($this->getTempDir(), '/valid_backup_filename__20211101141254.sql');
```
Comment (fritzmg) [inline core-bundle/tests/Doctrine/Backup/BackupTest.php:87]: Same issue here, the backup path will contain forward slashes, while the bath returned from the backup instance will contain backwards slashes on Windows. Using `Path::join` will automatically normalize the path.

File: core-bundle/tests/Doctrine/Backup/DumperTest.php [hunk 3]
```diff
@@ -61,7 +61,7 @@ public function testSuccessfulDump(array $tables, array $queries, string $expect
 
         $dumper->dump($connection, $config);
 
-        $this->assertSame($expectedDump, file_get_contents($backup->getFilepath()));
+        $this->assertSame($expectedDump, preg_replace('~\R~u', "\n", file_get_contents($backup->getFilepath())));
```
Comment (fritzmg) [inline core-bundle/tests/Doctrine/Backup/DumperTest.php:64]: This replaces all line endings with `\n` (which is used for the comparison).

File: core-bundle/tests/Doctrine/Backup/BackupTest.php (hunk 2 above)
Comment (m-vo) [inline core-bundle/tests/Doctrine/Backup/BackupTest.php]:
```suggestion
        return Path::join($this->getTempDir(), 'valid_backup_filename__20211101141254.sql');
```

File: core-bundle/tests/Doctrine/Backup/BackupTest.php [hunk 4]
```diff
@@ -83,6 +84,6 @@ public function invalidFileNameProvider(): \Generator
 
     private function getValidBackupPath(): string
     {
-        return $this->getTempDir().'/valid_backup_filename__20211101141254.sql';
+        return Path::join($this->getTempDir(), 'valid_backup_filename__20211101141254.sql');
```
Comment (fritzmg) [inline core-bundle/tests/Doctrine/Backup/BackupTest.php:87]: Similar issue here, the temp path will contain backwards slashes under Windows, while the path returned from the backup instance will contain only forward slashes now. Using `Path::join` will automatically normalize the path.

**Comment by Toflar** [comment]
Thanks for the fixes and the explanations ❤️

**Pull request merged**